kubernetes/data_source_kubernetes_deployment.go

```diff
@@ -0,0 +1,31 @@
+package kubernetes
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func dataSourceKubernetesDeployment() *schema.Resource {
+	dsSchema := datasourceSchemaFromResourceSchema(resourceKubernetesDeployment().Schema)
+
+	addRequiredFieldsToSchema(dsSchema, "metadata")
+	addRequiredFieldsToSchema(dsSchema["metadata"].Elem.(*schema.Resource).Schema, "name")
```

Comment:
@sl1pm4t not sure is this is the best way to be doing this...

Reply:
Ahh yeah, the heavily nested K8s resource schema's do complicate things a bit.
I'll have a look.

Reply:
I think that's the best way to handle this for now. Good job.